sdb/commands/member.py

```diff
@@ -31,16 +32,145 @@ class Member(sdb.Command):
     def _init_argparse(self, parser: argparse.ArgumentParser) -> None:
         parser.add_argument("members", nargs="+", metavar="<member>")
 
-    def call(self, objs: Iterable[drgn.Object]) -> Iterable[drgn.Object]:
-        for obj in objs:
-            for member in self.args.members:
+    @staticmethod
+    def _lex_member_tokens(member_expr: str) -> List[str]:
+        return list(filter(None, re.split(r'(\.|->|\[|\])', member_expr)))
+
+    def _parse_member_tokens(self, tokens: List[str]) -> List[Tuple[str, str]]:
+        terms = []
+        while tokens:
+
+            is_index = False
+            identifier = ""
+            idx = ""
+            if tokens[0] == "." or tokens[0] == "->":
+                sep = tokens[0]
+                identifier = tokens[1]
```

Comment:
I managed to trigger a crash on this line while experimenting with this change:
```
> spa | member spa_zio_taskq[0][0].stqs_taskq->
Traceback (most recent call last):
  File "/usr/local/bin/sdb", line 11, in <module>
    load_entry_point('sdb==0.1.0', 'console_scripts', 'sdb')()
  File "/usr/local/lib/python3.6/dist-packages/sdb-0.1.0-py3.6.egg/sdb/internal/cli.py", line 216, in main
    repl.start_session()
  File "/usr/local/lib/python3.6/dist-packages/sdb-0.1.0-py3.6.egg/sdb/internal/repl.py", line 114, in start_session
    _ = self.eval_cmd(line)
  File "/usr/local/lib/python3.6/dist-packages/sdb-0.1.0-py3.6.egg/sdb/internal/repl.py", line 87, in eval_cmd
    for obj in objs:
  File "/usr/local/lib/python3.6/dist-packages/sdb-0.1.0-py3.6.egg/sdb/__init__.py", line 180, in invoke
    yield from execute_pipeline(prog, first_input, pipeline)
  File "/usr/local/lib/python3.6/dist-packages/sdb-0.1.0-py3.6.egg/sdb/__init__.py", line 77, in execute_pipeline
    yield from pipeline[-1].call(this_input)
  File "/usr/local/lib/python3.6/dist-packages/sdb-0.1.0-py3.6.egg/sdb/commands/member.py", line 160, in call
    terms = self._parse_member_tokens(tokens)
  File "/usr/local/lib/python3.6/dist-packages/sdb-0.1.0-py3.6.egg/sdb/commands/member.py", line 48, in _parse_member_tokens
    identifier = tokens[1]
IndexError: list index out of range
```

Reply:
Thanks for giving a test-drive and finding this. I added 2 test cases and fixed it locally. Will push it once I fix the rest of the issues.